src/support/bits.cpp

```diff
@@ -158,6 +158,11 @@ int ceilLog2(uint32_t v) { return 32 - countLeadingZeroes(v - 1); }
 int ceilLog2(uint64_t v) { return 64 - countLeadingZeroes(v - 1); }
 
 bool isPowerOf2Float(float v) {
+  // Power of two float points should have zero significand part
+  // so here we just mask exponent range of "v" and compare it with
+  // unmasked input value and if both equal we our value is power of
+  // two. Also we reject all values which less of minmal possible
+  // power of two or maximum possible poser of two.
```

Comment:
```suggestion
  // Power of two floating points should have zero as their significands,
  // so here we just mask the exponent range of "v" and compare it with the
  // unmasked input value. If they are equal, our value is a power of
  // two. Also, we reject all values which are less than the minimal possible
  // power of two or greater than the maximum possible power of two.
```

Why do we also need to compare with the minimum and maximum powers of two? Shouldn't checking that the significand is 0 be enough?

Do we also need to explicitly check for NaN inputs?

Reply:
Why do we also need to compare with the minimum and maximum powers of two? Shouldn't checking that the significand is 0 be enough?

We need reject zeros and infinity / NaN range anyway so I decided create little bit stricter bounds. Anyway outside  this range [MIN_POT; MAX_POT] we will always get false result